src/core/lib/promise/loop.h

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_CORE_LIB_PROMISE_LOOP_H
+#define GRPC_CORE_LIB_PROMISE_LOOP_H
+
+#include <grpc/impl/codegen/port_platform.h>
+
+#include "absl/types/variant.h"
+#include "src/core/lib/promise/detail/promise_factory.h"
+
+namespace grpc_core {
+
+// Special type - signals to loop to take another iteration, instead of
+// finishing
+struct Continue {};
```

Comment:
nit, optional: `::grpc_core::Continue` seems like a very general name to put in such a broadly-used namespace. Consider either naming this something specific to Promises, or having it in a different namespace to avoid ambiguity. caveat: general c++ recommendation is for shallow namespace nesting.

Reply:
Ack on it being general, but I think that's a positive here: it's mirroring a language feature that's spelled similarly, in a way that few other types would be.

Reply:
Ah, I see. If `::grpc_core::Continue` becomes something we'd reach for in this codebase as much as we'd otherwise reach for `continue`, the namespace/generality fits pretty well.